File: GAMEOFLIFEJAVA/gameoflife/organisms/plants/Guarana.java
package gameoflife.organisms.plants;

import gameoflife.organisms.Organism;
import gameoflife.structs.Vector;

public class Guarana extends Plant {

    public Guarana(Vector position){
        super("+", 0, position);
    }

    @Override
    public void Collision(Organism other){

        other.AddStrength(3);
        world.Log(other.getClass().getSimpleName()+" got strength from eating a guarana");
        this.Kill();
    }
    
}

File: GAMEOFLIFE/Guarana.cpp
#include "Guarana.h"
#include "World.h"

Guarana::Guarana(int x, int y) : Plant('%', 9, x, y, 0, 0) {
}

void Guarana::Collision(Organism* other) {
	world->AddLog(this, "strengthens an animal", 1);
	other->Strengthen();
	Kill();
}

Organism* Guarana::Child() {
	return new Guarana(*this);
}